libraries/AP_Proximity/AP_Proximity_Backend.cpp

```diff
@@ -248,40 +248,53 @@ bool AP_Proximity_Backend::ignore_reading(uint16_t angle_deg) const
 }
 
 // returns true if database is ready to be pushed to and all cached data is ready
-bool AP_Proximity_Backend::database_prepare_for_push(Vector2f &current_pos, float &current_heading)
+bool AP_Proximity_Backend::database_prepare_for_push(Vector3f &current_pos, Matrix3f &body_to_ned)
 {
     AP_OADatabase *oaDb = AP::oadatabase();
     if (oaDb == nullptr || !oaDb->healthy()) {
         return false;
     }
 
-    if (!AP::ahrs().get_relative_position_NE_origin(current_pos)) {
+    if (!AP::ahrs().get_relative_position_NED_origin(current_pos)) {
         return false;
     }
 
-    current_heading = AP::ahrs().yaw_sensor * 0.01f;
+    body_to_ned = AP::ahrs().get_rotation_body_to_ned();
+
     return true;
 }
 
 // update Object Avoidance database with Earth-frame point
 void AP_Proximity_Backend::database_push(float angle, float distance)
 {
-    Vector2f current_pos;
-    float current_heading;
-    if (database_prepare_for_push(current_pos, current_heading)) {
-        database_push(angle, distance, AP_HAL::millis(), current_pos, current_heading);
+    Vector3f current_pos;
+    Matrix3f body_to_ned;
+
+    if (database_prepare_for_push(current_pos, body_to_ned)) {
+        database_push(angle, distance, AP_HAL::millis(), current_pos, body_to_ned);
     }
 }
 
 // update Object Avoidance database with Earth-frame point
-void AP_Proximity_Backend::database_push(float angle, float distance, uint32_t timestamp_ms, const Vector2f &current_pos, float current_heading)
+void AP_Proximity_Backend::database_push(float angle, float distance, uint32_t timestamp_ms, const Vector3f &current_pos, Matrix3f body_to_ned)
```

Comment:
we should probably rename this current_pos argument to either current_pos_neu or current_pos_ned.

We should also pass the body-toi_ned as a const reference.  i.e. const Matrix3f &body_to_ned.